### PR TITLE
django 1.10

### PR DESCRIPTION
**Pull request by nickpell**
opening for tests

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ install:
 before_script:
   - docker version
   - docker-compose version
-script: "scripts/docker test --noinput --stop --verbosity=2 --divide-depth=1"
+script: "scripts/docker test --noinput --verbosity=2 --divide-depth=1"
```
Comment (millerdev) [inline .travis.yml]: Curious why `--stop` was removed?

Comment (nickpell) [inline .travis.yml]: just for testing this PR so I can find >1 bug per iteration on travis

Comment (nickpell) [inline .travis.yml]: will remove before this PR gets merged

**Comment by snopoke** [comment]
@nickpell just wondering if it's a good idea to upgrade to a non-LTS release? The next LTS release (1.11) is due in April 2017.

**Comment by nickpell** [comment]
My two cents: It's easier to manage upgrading 1.8 -> 1.9 -> 1.10 -> 1.11 than 1.8 -> 1.11.  The 1.9 upgrade is ready to go whenever we pull the trigger.  As long as we get onto 1.10 by April 2017 we'll receive security and major bug fixes until December 2017.  I figure that should be enough time to get onto 1.11.

@snopoke

**Comment by millerdev** [comment]
FWIW, I'm with @nickpell on this. It's usually a lot easier to upgrade pervasive libraries like Django more frequently rather than waiting between each LTS release. Upgrading sooner means recommended new code patterns can be used sooner, and theoretically less code needs to be visited during each upgrade effort as a result.

**Comment by snopoke** [comment]
I generally agree with those arguments and am happy for us to proceed, just wanted to make sure we were being intentional about it.

File: corehq/ex-submodules/casexml/apps/case/tests/util.py [hunk 2]
```diff
@@ -207,3 +207,7 @@ def delete_all_xforms():
 @unit_testing_only
 def delete_all_sync_logs():
     FormProcessorTestUtils.delete_all_sync_logs()
+
+@unit_testing_only
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/tests/util.py]: E302 expected 2 blank lines, found 1

**Comment by nickpell** [comment]
Fixes all known incompatibilities and all the bugs found on staging in the past month.  Planning to release Wed morning EST.

**Comment by snopoke** [comment]
Sorry to put a damper on this but we need to hold off on merging this until we can sort out this issue:
* https://stackoverflow.com/questions/40851475/pass-django-csrf-token-to-angular-with-csrf-cookie-httponly

For NIC security audit we have to change CSRF token to HTTP-ONLY which means that it isn't accessible to JS. The workaround for that is this: http://stackoverflow.com/a/30656040/632517 but it doesn't work in Django 1.10 because it resets the CSRF token on every request.

**Comment by snopoke** [comment]
Other issues described here: https://security.stackexchange.com/a/22936/141942:

> it costs you in terms of usability: with only one token valid at once, the user will not be able to navigate the webapp normally. For example if they hit the 'back' button and submit the form with new values, the submission will fail, and likely greet them with some hostile error message. If they try to open a resource in a second tab, they'll find the session randomly breaks in one or both tabs.

**Comment by millerdev** [comment]
@snopoke on reviewing [the CSRF docs for Django 1.10](https://docs.djangoproject.com/en/1.10/ref/csrf/#how-it-works) as well as [the issue](https://code.djangoproject.com/ticket/20869) and [the commit that made the value sent to the client change on every request](https://github.com/django/django/commit/5112e65ef2df1dbb95ff83026b6a962fb2688661), it looks like the server value of the CSRF token is constant for the duration of the session, but the value sent to the client is scrambled with a random salt to protect it from BREACH attacks. This means it is protected against BREACH _and_ it has none of the usability issues of changing the token on every request. My conclusion from this is that we should upgrade to Django 1.10 as soon as possible since it has a superior CSRF protection implementation than older versions of Django.

**Pull request merged**